### PR TITLE
Fixing #3624 - Readline crash with @{<tab>

### PR DESCRIPTION
**Pull request by powercode**
Fix https://github.com/PowerShell/PowerShell/issues/3624

### CODE REVIEW

File: src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs [hunk 1]
```diff
@@ -442,6 +442,14 @@ internal List<CompletionResult> GetResultHelper(CompletionContext completionCont
                             result = GetResultForEnumPropertyValueOfDSCResource(completionContext, string.Empty, ref replacementIndex, ref replacementLength, out unused);
                         }
                         break;
+                    case TokenKind.AtCurly:
+                        // Handle scenarios such as Sort-Object @{<tab> || gci | Format-Table @{
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionAnalysis.cs]: > Sort-Object @{<tab> || gci | Format-Table @{

minor comment: can we change this to `'Sort-Object @{<tab>' or 'gci | Format-Table @{'`? I think it's more readable.

File: test/powershell/Host/TabCompletion/BugFix.Tests.ps1 [hunk 2]
```diff
@@ -54,4 +54,13 @@ Describe "Tab completion bug fix" -Tags "CI" {
            $result.CompletionMatches[1].CompletionText | Should Be $DatetimeProperties[1].Name # DateTime
        }
     }
+                
+    It "Issue#3628 - 'Sort-Object @{<tab> should work" {
```
Comment (daxian-dbw) [inline test/powershell/Host/TabCompletion/BugFix.Tests.ps1]: > 'Sort-Object @{<tab> should work

Minor comment: did you miss a single quote? :)

**Comment by powercode** [comment]
@daxian-dbw Thx for the review. Issues addressed.

**Comment by iSazonov** [comment]
LGTM.

@powercode I added link on Issue in description. The Issue will be auto closed after merge. And it is useful to see a link to the PR in the Issue.

**Comment by powercode** [comment]
@iSazonov I thought the number in the title did so, but apparently I was incorrect :)

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**